### DISCUSSION TITLE
Best way to run Symfony behind a reverse proxy

### DISCUSSION THREAD

**Discussion by Brammm** [symfony/symfony#42728] [Q&A]
We are trying to [strangle fig](https://martinfowler.com/bliki/StranglerFigApplication.html) an old application by slowly building new parts on top of it in a new Symfony application. Our new app will run on the same domain (example.com) as the old app, but we'll use nginx/our load balancer to proxy requests for the new app to a different container than the existing app with a url prefix (example.com/v2/reporting). This how the nginx server settings look like:
```
        location /v2/reporting {
            rewrite ^/v2/reporting/?(.*)$ /$1  break;
            proxy_pass http://127.0.0.1:8888;
        }
```

Currently, we have it set up so that nginx does not overwrite any headers, but I use a kernel.request event listener to overwrite the RequestContext (appUrl being set from an environment variable to https://example.com/v2/reporting):

```php
final class RouteContextEventSubscriber implements EventSubscriberInterface
{
    public function __construct(
        private RouterInterface $router,
        private string $appUrl
    ) {
    }

    /**
     * @inheritDoc
     */
    public static function getSubscribedEvents()
    {
        return [KernelEvents::REQUEST => 'onRequest'];
    }

    public function onRequest(RequestEvent $event): void
    {
        if (! $event->isMainRequest()) {
            return;
        }

        $this->router->setContext(RequestContext::fromUri($this->appUrl));
    }
}
```

Furthermore, the same APP_URL env var is used to set the default_uri for the router and the assets base_url. 

```yaml
# routing.yaml
framework:
    router:
        utf8: true
        default_uri: '%env(APP_URL)%'
    assets:
        base_urls:
            - '%env(APP_URL)%'
```

This works almost perfectly (currently still dealing with [an issue](https://github.com/symfony/symfony/discussions/42720) where fonts loaded in a css file and built with Webpack Encore don't load).

Reading the [proxies documentation](https://symfony.com/doc/current/deployment/proxies.html) however, I've come to realize we could also set up our nginx proxy pass to forward the host header and not strip off the url prefix. But I'm not sure that's a better approach. Has anyone done this before and what setup did you use?

**Comment by chyshyya** [comment]
I'm using this config without any patches to symfony app:

```
location ~* ^\/v2\/reporting.+?\.(ttf|js|css)${
  try_files /78ebea3e547ccaf7815a77c74ed6a5a8.htm @proxy;
}

location /v2/reporting {
  try_files /78ebea3e547ccaf7815a77c74ed6a5a8.htm @proxy;
}

location @proxy {
  proxy_set_header Host $http_host;
  proxy_set_header X-Real-IP $remote_addr;
  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header X-Forwarded-Proto $scheme;
  proxy_pass 127.0.0.1:8888;
}
```
But your css font urls must be prefixed with `/v2/reporting` with webpack url-resolver as I give you an answer in [discussion](https://github.com/symfony/symfony/discussions/42720).

**Reply by chyshyya** [reply]
But we did your job with subdomains like v2.example.com. This will be easy to proxy domain and transfer between domains. Just an advise. Nginx config i gave you do this like a charm.
